src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java

```diff
@@ -1001,6 +1016,20 @@ public ResultSet getProcedures(String catalog, String schemaPattern, String proc
 
     ResultSet resultSet =
         executeAndReturnEmptyResultIfNotFound(statement, showProcedureCommand, GET_PROCEDURES);
+    String queryId;
+    try {
```

Comment:
The reason for the try/catch is that the ResultSet can either come back as a SnowflakeResultSet or a SnowflakeDatabaseMetaDataResultSet. The latter is only used to return empty result sets with no real data. It cannot be unwrapped from the SnowflakeResultSet class and does not have a query ID associated with it anyway.

So if an exception is returned saying DatabaseMetaDataResultSet cannot be unwrapped from SnowflakeResultSet, an empty query ID can be returned.

Reply:
I would use `instanceof` since it is faster than try catch. https://stackoverflow.com/questions/20431614/throw-and-catch-an-exception-or-use-instanceof

Reply:
Also, you can create a helper function to reuse the code.